src/DynamoCoreWpf/Controls/StartPage.xaml.cs

```diff
@@ -419,7 +419,15 @@ private void OnRecentFilesChanged(object sender, NotifyCollectionChangedEventArg
             }
             else if (e.Action == NotifyCollectionChangedAction.Remove)
             {
-                recentFiles.RemoveRange(e.OldStartingIndex, recentFiles.Count - e.OldStartingIndex);
+                var removedItemsCount = e.OldItems?.Count ?? 0;
+                if (recentFiles.Count > e.OldStartingIndex && recentFiles.Count >= removedItemsCount)
+                {
+                    recentFiles.RemoveRange(e.OldStartingIndex, removedItemsCount);
```

Comment:
What if you're starting to remove items from the starting index, but the (index from where you're starting to remove items) + `removedItemsCount` is greater than the total count of `recentFiles`, this could lead to an index out of range error, can't it?

Reply:
The remove range code is called when the RecentFiles collection is changed, specifically when the cache size goes beyond the allowed limit. This happen in 2 cases, user opens a 11th file when the limit is set to 10, in this case the new file is added to the list, size goes to 11, the last element at index 10 is removed, so the OldStartingIndex is 10, and removedItemsCount is 1.
The other case is when recent files limit is reduced from preferences, in this case lets say user updates the limit to 5, the OldStartingIndex is now 5, removedItemsCount is also 5.
Update method is here: https://github.com/DynamoDS/Dynamo/blob/165f1d29c500c527b0170eb3c0ba341277d4748e/src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs#L1882